helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -1312,12 +1312,9 @@ private Message markReadMessage(Message message, NotificationContext context,
 
   private void updateUnprocessableMessage(Message message, Exception exception, String errorMsg,
       HelixManager manager) {
-    String error = "Message " + message.getMsgId() + " cannot be processed: " + message.getRecord();
     if (exception != null) {
-      LOG.error(error, exception);
-      _statusUpdateUtil.logError(message, HelixStateMachineEngine.class, exception, error, manager);
+      _statusUpdateUtil.logError(message, HelixStateMachineEngine.class, exception, errorMsg, manager);
     } else {
-      LOG.error(error + errorMsg);
       _statusUpdateUtil.logError(message, HelixStateMachineEngine.class, errorMsg, manager);
     }
```

Comment:
Thanks for the review. _statusUpdateUtil writes error messages to ZK and also log error message. 

```
  void publishErrorRecord(ZNRecord record, String instanceName, String updateSubPath,
      String updateKey, String sessionId, HelixDataAccessor accessor, boolean isController) {
    _logger.error("StatusUpdate Error record: {}", record);
```

Reply:
I see. I think the two logs serve different purposes though. Is there any way we could log it once and be done for a given message? 

Here, the first log tells us that the message couldn't be processed, but the second log is a mere log of the record of what's being written to ZK.

You could also keep that `String error = "Message " + message.getMsgId() + " cannot be processed";` since it is enough since the record will be logged anyway.

Reply:
MessageId is included in the `message` object and will be logged as a field of `record `.
Also the specific reasons for failing to handling the message is put to the record as `AdditionalInfo`. 

The information that state "message can not be processed" is duplicate.

Reply:
I think we just need to remove the regular log behavior in _statusUpdateUtil. So it only writes to file.

Reply:
There are many usage of  _statusUpdateUtil.logError(). If we remove the log behavior in _statusUpdateUtil, then we need to add an info or warning log to every caller. 
I would prefer to keep the log at one place (this means remove the duplicate log at other place as well...).